code/game/machinery/dance_machine.dm

```diff
@@ -436,26 +436,25 @@
 	if(world.time < stop && active)
 		var/sound/song_played = sound(selection.song_path)
 
-		for(var/mob/M in range(10,src))
-			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
-				continue
-			if(!(M in rangers))
-				rangers[M] = TRUE
-				M.playsound_local(get_turf(M), null, 100, channel = CHANNEL_JUKEBOX, S = song_played)
 		for(var/mob/L in rangers)
 			if(get_dist(src,L) > 10)
 				rangers -= L
 				if(!L || !L.client)
 					continue
 				L.stop_sound_channel(CHANNEL_JUKEBOX)
+		for(var/mob/M in range(10,src))
```

Comment:
I'm only half paying attention to this file, but why're you moving this?

Reply:
I just wanted to reduce the amount of possible iteration at
https://github.com/BeeStation/BeeStation-Hornet/blob/8fd9822595e62899093087d34cdadaba514682e6/code/game/machinery/dance_machine.dm#L448
since this change will prevent that `in` operator from iterating through occasional mobs in `ranger` that will be invalidated right after that loop because ranger just got out of the range. (it should have no change in functionality since the proc already deals with out of range mobs adequately)